az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -80,6 +86,12 @@ public class HadoopJobUtils {
   public static final String MAPREDUCE_JOB_OTHER_NAMENODES = "mapreduce.job.hdfs-servers";
   // MapReduce config for mapreduce job tags
   public static final String MAPREDUCE_JOB_TAGS = "mapreduce.job.tags";
+
+  public static final String YARN_KILL_VERSION = "yarn.kill.version";
+  // values of the yarn kill version flag
+  public static final String YARN_KILL_LEGACY = "legacy";
+  public static final String YARN_KILL_USE_API_WITH_TOKEN = "api_with_token";
```

Comment:
For most features an appropriately named boolean flag is adequate. For example in this case it could be something like the property `enable.advanced.yarn.app.kill` or `enable.yarn.kill.with.token` with true/false values.
That also makes a bit easier for someone to reason about the code in future and eliminate the flag when not needed anymore. 

For now, we can retain this property as-is as it will be removed in the near future.

Reply:
I am using string because it could be non-binary, e.g. when we substitute getAppFromLog with getAppWithApi, we can have a 3rd option.

good advice on naming